src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,1430 @@
+        }
+
+        [Fact]
+        public void InterfaceImpl_03()
+        {
+            const string text = @"
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+interface I
+{
+    void M([AllowNull] string s1);
+}
+
+partial class C : I
+{
+    public partial void M(string s1); // 1
+}
+
+partial class C
+{
+    public partial void M(string s1)
+    {
+        s1.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(new[] { text, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (13,25): warning CS8767: Nullability of reference types in type of parameter 's1' of 'void C.M(string s1)' doesn't match implicitly implemented member 'void I.M(string s1)' (possibly because of nullability attributes).
+                //     public partial void M(string s1);
+                Diagnostic(ErrorCode.WRN_TopLevelNullabilityMismatchInParameterTypeOnImplicitImplementation, "M").WithArguments("s1", "void C.M(string s1)", "void I.M(string s1)").WithLocation(13, 25));
+        }
+
+        [Fact]
+        public void InterfaceImpl_04()
+        {
+            const string text = @"
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+interface I
+{
+    void M([AllowNull] string s1, [AllowNull] string s2);
+}
+
+partial class C : I
+{
+    public partial void M([AllowNull] string s1, string s2);
+}
+
+partial class C
+{
+    public partial void M(string s1, [AllowNull] string s2)
+    {
+        s1.ToString(); // 1
+        s2.ToString(); // 2
+    }
+}
+";
+            // PROTOTYPE: should it be allowed to specify duplicate nullability attributes?
+            // if not, should there be some convention about which part should contain the nullability attributes?
+            var comp = CreateCompilation(new[] { text, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (20,9): warning CS8602: Dereference of a possibly null reference.
+                //         s1.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s1").WithLocation(20, 9),
+                // (21,9): warning CS8602: Dereference of a possibly null reference.
+                //         s2.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s2").WithLocation(21, 9));
+        }
+
+        [Fact]
+        public void InterfaceImpl_05()
+        {
+            const string text = @"
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+interface I
+{
+    void M([AllowNull] string s1);
+}
+
+partial class C : I
+{
+    public partial void M([AllowNull] string s1);
+}
+
+partial class C
+{
+    public partial void M([AllowNull] string s1) // 1
+    {
+        s1.ToString(); // 2
+    }
+}
+";
+            // PROTOTYPE: should it be allowed to specify duplicate nullability attributes?
```

Comment:
>should it be allowed to specify duplicate nullability attributes? [](start = 26, length = 65)

If we allow nullability attributes on both parts, we should allow attributes on both parts for all attributes.

Reply:
It is feeling like maybe we should just ship the current rules and see how people feel about it. Adding special allowances for duplicate attributes in partials seems like something you can't just implement speculatively, we have to see how people end up using/experiencing it.